upf/action.py

```diff
@@ -163,3 +159,6 @@ def add_decrease_effect(self, fluent: Union[FNode, 'upf.Fluent'],
         if not self._env.type_checker.is_compatible_type(fluent_exp, value_exp):
             raise UPFTypeError('Action effect has not compatible types!')
         self._effects.append(Effect(fluent_exp, value_exp, condition_exp, kind = DECREASE))
+
+    def add_effect(self, effect: Effect):
+        self._effects.append(effect)
```

Comment:
After thinking a little about that, I think it is better to remove this method and keep only the original 3 methods to add new effects.